crates/rpc/src/v04/method/add_deploy_account_transaction.rs

```diff
@@ -98,30 +100,92 @@ pub async fn add_deploy_account_transaction(
     context: RpcContext,
     input: AddDeployAccountTransactionInput,
 ) -> Result<AddDeployAccountTransactionOutput, AddDeployAccountTransactionError> {
+    use starknet_gateway_types::request::add_transaction;
+
     let Transaction::DeployAccount(tx) = input.deploy_account_transaction;
-    let response = context
-        .sequencer
-        .add_deploy_account(
-            tx.version,
-            tx.max_fee,
-            tx.signature,
-            tx.nonce,
-            tx.contract_address_salt,
-            tx.class_hash,
-            tx.constructor_calldata,
-        )
-        .await?;
-
-    Ok(AddDeployAccountTransactionOutput {
-        transaction_hash: response.transaction_hash,
-        contract_address: response.address,
-    })
+
+    match tx {
+        BroadcastedDeployAccountTransaction::V0V1(
+            tx @ BroadcastedDeployAccountTransactionV0V1 { version, .. },
+        ) if version.without_query_version() == 0 => {
```

Comment:
I wonder if there isn't a better abstraction for base version + query flag e.g. an enum + bool which would make this safer to use. Problem for another day though, this is fine for now.

Reply:
I'm sure there is a better representation -- fully agreed.